### ISSUE TITLE
add integration with OneDrive

### ISSUE THREAD

**Issue by imnbwd** [Microsoft/WindowsTemplateStudio#362]
add integration with OneDrive, so that the local data file can be automatically sync to OneDrive

**Comment by ralarcon** [comment]
Hi @imnbwd, do you mean to use OneDrive as a generic roaming data store? Currently, Settings feature provide a way to access the roaming profile (with the limits in storage size the it has). Are you thinking in have increased storage size?

Thanks.

**Comment by imnbwd** [comment]
Yes, what I mean is I want there is a data sync option, which is OneDrive. That is, the user can sync the app data files to OneDrive.

Suppose it works like this way: If I check that option in the wizard, OneDrive package (maybe also include the related authentication packages for OneDrive) will be added to the project, a basic class containing some methods(such as upload/download/delete file, check file exist ...) is also provided, maybe a page consuming that class is also provided. that's all. 
There are a few apps possessing such a feature, just like these 2 apps of Microsoft: Office Lens, Fresh Paint.

**Comment by mrlacey** [comment]
@imnbwd We're going to be looking at doing more with different data stores in a future version. Looking at things like databases, web content, etc.

The [community toolkit has a wrapper for calling the OneDrive Service](https://uwpcommunitytoolkit.readthedocs.io/en/master/services/OneDrive/) so you can do things like authenticating and file access. It doesn't do any automatic syncing but would including this service be sufficient for your needs?

**Comment by crutkas** [comment]
I'm still thinking how this would be added as it feels very tied to your own apps logic.  A background task we can wire up, this feels very tied to how an app would want to implement the feature.  

There also is the ApplicationData.RoamingFolder, while limited space, does roam with your using.
https://msdn.microsoft.com/en-us/library/windows/apps/hh465094.aspx

Here is the screenshot from Fresh Paint.
![capture](https://cloud.githubusercontent.com/assets/1462282/26214369/f6ff2db6-3bb0-11e7-8a00-0df4d542be2b.PNG)

**Comment by crutkas** [comment]
@imnbwd can you provide more information here on your scenario?  Not just implementation.  They are storing large files, not settings.

**Comment by Javier118** [comment]
Optional Login can be used with oneDrive
